src/decode.h

```diff
@@ -250,7 +276,68 @@ static inline uint32_t sign_extend_b(const uint32_t x)
     return (int32_t) ((int8_t) x);
 }
 
+/* hash function is used when mapping address into the block map */
+static inline uint32_t hash(size_t k)
+{
+    k ^= k << 21;
+    k ^= k >> 17;
+#if (SIZE_MAX > 0xFFFFFFFF)
+    k ^= k >> 35;
+    k ^= k >> 51;
+#endif
+    return k;
+}
+
+/* allocate a basic block */
+static inline struct block *block_alloc(const uint8_t bits)
```

Comment:
`block_alloc` should not appear in `decode.h`. Make the header as simple as possible.